tests/tensor/rewriting/test_linalg.py

```diff
@@ -412,9 +414,9 @@ def test_det_diag_from_eye_mul(shape):
     if len(shape) == 0:
         x_test = np.array(np.random.rand()).astype(config.floatX)
     elif len(shape) == 1:
-        x_test = np.random.rand(7).astype(config.floatX)
+        x_test = np.random.rand(*shape).astype(config.floatX)
     else:
-        x_test = np.random.rand(7, 7).astype(config.floatX)
+        x_test = np.random.rand(*shape).astype(config.floatX)
     det_val = f_det(x_test)
```

Comment:
You should use a manually computed reference value, because your `f_det` will have the standard rewrites (including the one you're testing) and if the rewrite changes the meaning of the expression (i.e., has a bug) you won't catch it. Alternatively you can compile a function without any rewrites by passing something like `mode=Mode(linker="py", optimizer=None)` to `f_det`

Reply:
ohhh so the default function `f_det` also gets the rewrite applied to it because i have added it already. so i should use the `mode` thing to make it compile directly without using the rewrite?

Reply:
Yes, or get the expected "dumb" result with numpy operations directly, whichever you prefer